### PR TITLE
#2413 Do not fail on empty template

### PR DESCRIPTION
**Pull request by BALEHOK**
Re #2413

### CODE REVIEW

File: src/runtime/mapArgs.ts [hunk 1]
```diff
@@ -97,6 +97,12 @@ export async function renderExplicit(
   options: RendererOptions
 ): Promise<unknown> {
   if (isTemplateExpression(config)) {
+    // This check is added to prevent exceptions when rendering a faulty template
+    // see https://github.com/pixiebrix/pixiebrix-extension/issues/2413
+    if (config.__value__ == null) {
```
Comment (twschiller) [inline src/runtime/mapArgs.ts:102]: - [ ] This is OK for text templates (nunjucks, etc.) but is the wrong behavior for var. For var we would need this to return null

**Review by twschiller — CHANGES_REQUESTED** [review]
- [ ] Don't return empty string for "var"

File: src/runtime/mapArgs.test.ts [hunk 2]
```diff
@@ -44,6 +44,18 @@ describe("renderExplicit", () => {
       foo: "bar!",
     });
   });
+
+  test("doesn't fail on empty template", async () => {
+    const rendered = await renderExplicit(
+      { foo: { __type__: "nunjucks", __value__: undefined } },
```
Comment (twschiller) [inline src/runtime/mapArgs.test.ts:50]: Use `each` to cover the other template types

File: src/runtime/mapArgs.ts (hunk 1 above)
Comment (BALEHOK) [reply src/runtime/mapArgs.ts:102]: var is a template... good catch

File: src/runtime/mapArgs.ts [hunk 3]
```diff
@@ -87,6 +87,16 @@ export function isTemplateExpression(
   );
 }
 
+/**
+ * Returns true if value represents an explicit template engine expression
+ * @see isExpression
+ */
+export function isVarExpression(
+  value: unknown
+): value is Expression<string, TemplateEngine> {
```
Comment (BLoe) [inline src/runtime/mapArgs.ts:96]: I think the type here can be `Expression<string, "var">`.

File: src/runtime/mapArgs.test.ts [hunk 4]
```diff
@@ -44,6 +44,24 @@ describe("renderExplicit", () => {
       foo: "bar!",
     });
   });
+
+  test.each([
+    ["mustache", { foo: "" }],
+    ["nunjucks", { foo: "" }],
+    ["handlebars", { foo: "" }],
+    ["var", {}],
```
Comment (twschiller) [inline src/runtime/mapArgs.test.ts:52]: NIT: add comment here that foo gets stripped out because the renderExplicit drops entries with nullish values?

**Review by twschiller — APPROVED** [review]

**Comment by twschiller** [comment]
👍  I'm good once Ben's good on this

**Review by BLoe — APPROVED** [review]

**Pull request merged**